app/main.py

```diff
@@ -1,2 +1,64 @@
+from typing import Any
+
+
+class Node:
+    def __init__(self, key: Any, value: Any, hash_value: int) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash_value
+        self.next = None
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.load_factor = 0.7
+        self.table = [None] * self.capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        current_index = self._hash_index(key)
+        current = self.table[current_index]
+        while current is not None:
+            if current.key == key:
+                current.value = value
+                return
+            current = current.next
+
+        new_node = Node(key, value, hash(key))
```

Comment:
Consider storing the hash value in a variable to avoid recalculating it multiple times. This can improve efficiency, especially if the hash function is complex.